### ISSUE TITLE
Adding a job using POST

### ISSUE THREAD

**Issue by chromodome** [inlife/nexrender#311]
First question:
using an AJAX POST request, where does one put the --secret header?

Second question:
I've edited parts of an HTML than @inlife did a while back using POST

```
<!DOCTYPE html>
<html>
  <head>
    <title>nexrender - form</title>
    <script
      src="https://code.jquery.com/jquery-3.3.1.min.js"
      crossorigin="anonymous"
    ></script>
    <link
      rel="stylesheet"
      type="text/css"
      href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css"
    />
  </head>
  <body>
    <div class="container">
      <form id="main" action="http://localhost:3000">
        <h2>Name and Start Time</h2>

        <div class="form-row">
          <div class="col">
            <input
              type="text"
              id="fName"
              class="form-control"
              name="fName"
              value="write something"
            />
          </div>
          <div class="col">
            <input
              type="text"
              id="fStart"
              class="form-control"
              name="fStart"
              value="2"
            />
          </div>
        </div>

        <h2>Your file name</h2>
        <div class="form-row">
          <div class="col">
            <input
              id="fFileName"
              class="form-control"
              name="fFileName"
              value="encoded"
            />
          </div>
        </div>

        <button type="submit" class="btn btn-primary">
          Submit
        </button>
      </form>
    </div>

    <script type="text/javascript">
      $("form#main").submit(function(e) {
        e.preventDefault();

        myJSON = {
          template: {
            src: "file:///c:/NodeJS_projects/nexrender/aep/nex-project.aep",
            composition: "main"
          },
          assets: [
            {
              type: "data",
              layerName: "dataName",
              property: "Source Text",
              value: "my Name"
            },
            {
              type: "data",
              layerName: "dataStart",
              property: "Source Text",
              value: "15"
            }
          ],
          actions: {
            postrender: [
              {
                module: "@nexrender/action-encode",
                preset: "mp4",
                output: ""
              },
              {
                module: "@nexrender/action-copy",
                input: "",
                output: "c:/NodeJS_projects/nexrender/output/"
              }
            ]
          }
        };

        myJSON.assets[0].value = $("#fName").val();
        myJSON.assets[1].value = $("#fStart").val();
        var tmpOutput =
          Math.random()
            .toString(36)
            .substring(2, 15) +
          Math.random()
            .toString(36)
            .substring(2, 15);
        myJSON.actions.postrender[0].output = myJSON.actions.postrender[1].input =
          tmpOutput + ".mp4";
        myJSON.actions.postrender[1].output = $("#fFileName").val() + ".mp4";

        //var formJSON = JSON.stringify(myJSON);
        var formJSON = myJSON;
      
        $.ajax({
          type: "POST",
          url: "http://localhost:3000/api/projects",
          dataType: "json",
          data: formJSON,

          success: function(result) {
            console.log(result);
          },
          error: function(error) {
            console.log(error);
          }
        });
      });
    </script>
  </body>
</html>
```
When i submit, nothing happens. Ther server doesn't pick up the job. The console logs a success with result=undefined

**Environment**
Running nexrender-server locally and the html is also local.

**Comment by chromodome** [comment]
Ok, part of the problem solved

it works if i submit the job to /api/v1/jobs like so:

```
$.ajax({
          type: "POST",
          url: "http://localhost:3000/api/v1/jobs",
          dataType: "json",
          data: formJSON,

          success: function(result) {
            console.log(result);
          },
          error: function(error) {
            console.log(error);
          }
        });
      });
```

I guess it's because i'm defining the job with template?

```
myJSON = {
          template: {
            src: "file:///c:/NodeJS_projects/nexrender/aep/nex-project.aep",
            composition: "main"
          }
}
```

2 more things not working for me:
1. Defining the **secret header** using POST
2. Solving browser CORS error. When I try to POST from domain1.com to the server on anotherdomain.com I get a "No Access-Control-Origin header is present on the requested resource".

I'm starting to understand what it is, but not how to solve it.

**Comment by inlife** [comment]
Hello,

You can check the module https://github.com/inlife/nexrender/tree/master/packages/nexrender-api for reference on how to provide the secret to the api. And if you are planning on using something more modern for the frontend part, like react/angular you can even just use that `@nexrender/api` module directly.

Regarding CORS, it might be indeed an issue, I'll take a look if our server-side takes any care of thecors instructions.

**Comment by chromodome** [comment]
Thanks @inlife
It's not really clear to me from the `@nexrender/api` where `--secret` should go.
I'm assuming it should be something like:
```
$.ajax({
          type: "POST",
          url: "http://localhost:3000/api/v1/jobs",
          dataType: "json",
          data: formJSON,
          headers: { "nexrender-secret": "mysecret" },
          success: function(result) {
            console.log(result);
          },
          error: function(error) {
            console.log(error);
          }
        });
      });
```
The only thing this does is throw a major CORS error, this time concerning preflight request (all locally).

![cors_error2](https://user-images.githubusercontent.com/60780857/74065175-f210e080-49fc-11ea-9437-4cf36c229384.jpg)

Any ideas what i'm getting myself into and how to navigate through all this?

Thanks

**Comment by inlife** [comment]
I'm working on fixing that cors issue.

**Comment by chromodome** [comment]
just FYI, it works fine without the ` headers: { "nexrender-secret": "mysecret" },` line and if obviously the server is set to no secret (I'm speaking about running locally in this case)

**Comment by inlife** [comment]
A new build is being built and published (`v1.11.4`), please try it out as soon as it becomes available to see if it fixes the issue.

**Comment by chromodome** [comment]
The CORS issue worked with POST. Thx @inlife 
I still haven't figured out how to send the `secret` with a `POST` request. If it should be done using a custom header then there's a new CORS error (caused by that header)
```
$.ajax({
          type: "POST",
          url: "http://localhost:3000/api/v1/jobs",
          headers: { "nexrender-secret": "mysecret" },
          dataType: "json",
          data: formJSON,
          ...
          
```

**Comment by inlife** [comment]
Honestly, haven't been using ajax in quite a long time now, don't remember all the quirks.
Check if this will help: https://gist.github.com/pithyless/1547408

>then there's a new CORS error (caused by that header)

Could you show the new error that appears?

**Comment by chromodome** [comment]
It's the one i attached 2 posts up. Here it is again.
![cors_error2](https://user-images.githubusercontent.com/60780857/74067184-4c13a500-4a01-11ea-8701-eefd21c1310f.jpg)

**Comment by inlife** [comment]
Does it still occur with the new updated version I published?

**Comment by chromodome** [comment]
Yes. Your newly published update `v1.11.4` solves the CORS policy error when adding a job with POST but without a `secret` . I have a feeling my approach to adding the secret (the syntax) with AJAX might be wrong.

**Comment by inlife** [comment]
Fixed in `v1.11.5`
